shared/protocol/src/main/java/io/pravega/shared/NameUtils.java

```diff
@@ -185,20 +200,60 @@ public static String getTransactionNameFromId(String parentStreamSegmentName, UU
     }
 
     /**
-     * Attempts to extract the name of the Parent StreamSegment for the given Transaction StreamSegment. This method returns a
-     * valid value only if the Transaction StreamSegmentName was generated using the generateTransactionStreamSegmentName method.
+     * Returns the transient name for a TransientSegment based on the name of the current Parent StreamSegment, and the transientId.
      *
-     * @param transactionName The name of the Transaction StreamSegment to extract the name of the Parent StreamSegment.
+     * @param parentStreamSegmentName The name of the Parent StreamSegment for this transient segment.
+     * @param writerId The Writer Id used to create the transient segment.
+     * @return The name of the Transient StreamSegmentId.
+     */
+    public static String getTransientNameFromId(String parentStreamSegmentName, UUID writerId) {
+        UUID random = UUID.randomUUID();
```

Comment:
Let's also write a test to make sure that transient segment names and transaction segment names are not confused by the new code or by the existing code (the one for transactions).